From c3040ffb82e2075334e26cf81c4bc7a119075f58 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Wed, 26 May 2021 06:50:56 +0300
Subject: [PATCH] [SW-43507] gaudi: add multi irq poll interfaces

Use multiple IRQ polling interfaces to prevent host IRQs overriding each
other.

The host IRQs are:
- PI update
- Halt machine
- INTS

Change-Id: I5dfaec88369b123fcc1ff253c8c37b9fecb5edb8
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index e04ed2b..9eda3cd 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -419,7 +419,9 @@
 	regs->hw_state = mmHW_STATE;
 	regs->kmd_msg_to_cpu = mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU;
 	regs->cpu_cmd_status_to_host = mmCPU_CMD_STATUS_TO_HOST;
-	regs->gic_host_irq_ctrl = mmGIC_HOST_IRQ_CTRL_POLL_REG;
+	regs->gic_host_pi_upd_irq = mmGIC_HOST_PI_UPD_IRQ_POLL_REG;
+	regs->gic_host_halt_irq = mmGIC_HOST_HALT_IRQ_POLL_REG;
+	regs->gic_host_ints_irq = mmGIC_HOST_INTS_IRQ_POLL_REG;
 	regs->gic_tpc_qm_irq_ctrl = mmGIC_TPC_QM_IRQ_CTRL_POLL_REG;
 	regs->gic_mme_qm_irq_ctrl = mmGIC_MME_QM_IRQ_CTRL_POLL_REG;
 	regs->gic_dma_qm_irq_ctrl = mmGIC_DMA_QM_IRQ_CTRL_POLL_REG;
